src/structural_properties.c

```diff
@@ -6006,36 +6006,50 @@ int igraph_i_avg_nearest_neighbor_degree_weighted(const igraph_t *graph,
  * Average nearest neighbor degree.
  *
  * Calculates the average degree of the neighbors for each vertex, and
- * optionally, the same quantity in the function of vertex degree.
+ * optionally, the same quantity as a function of the vertex degree.
  *
- * </para><para>For isolate vertices \p knn is set to
- * NaN. The same is done in \p knnk for vertex degrees that
+ * </para><para>
+ * For isolated vertices \p knn is set to NaN.
+ * The same is done in \p knnk for vertex degrees that
  * don't appear in the graph.
  *
- * \param graph The input graph, it can be directed but the
- *   directedness of the edges is ignored.
+ * </para><para>
+ * The weighted version computes a weighted average of the neighbor degrees as
+ *
+ * <code>k_nn_u = 1/s_u sum_u w_uv k_v</code>,
+ *
+ * where <code>s_u</code> is the sum of the incident edge weights of vertex \c u,
+ * i.e. its strength. The sum runs over the neighbors \c v of vertex \c u
+ * as indicated by \p mode (with appropriate weights)
+ * and <code>k_v</code> is the neighbors' degree, specified by \p neighbor_degree_mode.
+ *
+ * </para><para>
+ * Reference:
+ * A. Barrat, M. Barthélemy, R. Pastor-Satorras, and A. Vespignani,
+ * The architecture of complex weighted networks,
+ * Proc. Natl. Acad. Sci. USA 101, 3747 (2004).
+ * https://dx.doi.org/10.1073/pnas.0400087101
+ *
+ * \param graph The input graph. It may be directed.
  * \param vids The vertices for which the calculation is performed.
- * \param mode The neighbors over which is averaged.
- * \param neighbor_degree_mode The degree of the neighbors which is
- *   averaged.
+ * \param mode The type of neihbors to consider in directed graphs.
+ *   \c IGRAPH_OUT considers out-neighbours, \c IGRAPH_IN in-neighbours
```

Comment:
Here you use UK spelling, I believe we use US spelling throughout, so it should be `neighbors` instead of `neighbours`.

Reply:
Agree to keep US spelling, and I have already been trying to do so. This was a mistake (usually I use UK spelling so I have to make an extra effort here).

Reply:
Yeah, I have to same problem often: I tend to default to UK spelling.